File: coresuite-dotnet-core-console-vb/merger/AcroFormFill.vb
Imports ceTe.DynamicPDF.Merger

Public Class AcroFormFill
    Public Shared Sub Run(outputPdfPath As String)

        ' Create a document and set it's properties 
        Dim MyDocument As MergeDocument = New MergeDocument(Util.GetResourcePath("PDFs/fw9AcroForm_18.pdf"))
        MyDocument.Creator = "AcroFormFill.aspx"
        MyDocument.Author = "ceTe Software"
        MyDocument.Title = "W-9 AcroForm Filler"

        ' Get the imorted MyPage and set margins.
        Dim MyPage As ceTe.DynamicPDF.Page = MyDocument.Pages(0)

        ' Add content to the MyPage
        MyDocument.Form.Fields("topmostSubform[0].Page1[0].f1_1[0]").Value = "Any Company, Inc."
        MyDocument.Form.Fields("topmostSubform[0].Page1[0].f1_2[0]").Value = "Any Company"
        MyDocument.Form.Fields("topmostSubform[0].Page1[0].FederalClassification[0].c1_1[0]").Value = "1"
        MyDocument.Form.Fields("topmostSubform[0].Page1[0].Address[0].f1_7[0]").Value = "123 Main Street"
        MyDocument.Form.Fields("topmostSubform[0].Page1[0].Address[0].f1_8[0]").Value = "Washington, DC  22222"
        MyDocument.Form.Fields("topmostSubform[0].Page1[0].f1_9[0]").Value = "Any Requester"
        MyDocument.Form.Fields("topmostSubform[0].Page1[0].f1_10[0]").Value = "17288825617"
        Dim Ssn As String = "265748".Trim().Replace("-", "").Replace(" ", "")
        Dim Ein As String = "521234567".Trim().Replace("-", "").Replace(" ", "")
        If Ssn.Length >= 9 Then
            MyDocument.Form.Fields("topmostSubform[0].Page1[0].SSN[0].f1_11[0]").Value = Ssn.Substring(0, 1) + Ssn.Substring(1, 1) + Ssn.Substring(2, 1)
            MyDocument.Form.Fields("topmostSubform[0].Page1[0].SSN[0].f1_12[0]").Value = Ssn.Substring(3, 1) + Ssn.Substring(4, 1)
            MyDocument.Form.Fields("topmostSubform[0].Page1[0].SSN[0].f1_13[0]").Value = Ssn.Substring(5, 1) + Ssn.Substring(6, 1) + Ssn.Substring(7, 1) + Ssn.Substring(8, 1)
        ElseIf Ein.Length >= 9 Then
            MyDocument.Form.Fields("topmostSubform[0].Page1[0].EmployerID[0].f1_14[0]").Value = Ein.Substring(0, 1) + Ein.Substring(1, 1)
            MyDocument.Form.Fields("topmostSubform[0].Page1[0].EmployerID[0].f1_15[0]").Value = Ein.Substring(2, 1) + Ein.Substring(3, 1) + Ein.Substring(4, 1) + Ein.Substring(5, 1) + Ein.Substring(6, 1) + Ein.Substring(7, 1) + Ein.Substring(8, 1)

        End If

        ' Uncomment to make form read only
        'MyDocument.Form.IsReadOnly = True

        ' Outputs the W-9 to the current web MyPage
        MyDocument.Draw(outputPdfPath)


    End Sub
End Class

File: coresuite-dotnet-framework-console-cs/merger/AcroFormFill.cs
using ceTe.DynamicPDF.Merger;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace coresuite_dotnet_framework_console_cs.Merger
{
  public  class AcroFormFill
    {
        public static void Run(string outputPdfPath)
        {
            // Create a document and set it's properties
            MergeDocument document = new MergeDocument(Util.GetResourcePath("PDFs/fw9AcroForm_18.pdf"));
            document.Creator = "AcroFormFill.aspx";
            document.Author = "ceTe Software";
            document.Title = "W-9 AcroForm Filler";

            // Add content to the page
            document.Form.Fields["topmostSubform[0].Page1[0].f1_1[0]"].Value = "Any Company, Inc.";
            document.Form.Fields["topmostSubform[0].Page1[0].f1_2[0]"].Value = "Any Company";
            document.Form.Fields["topmostSubform[0].Page1[0].FederalClassification[0].c1_1[0]"].Value = "1";
            document.Form.Fields["topmostSubform[0].Page1[0].Address[0].f1_7[0]"].Value = "123 Main Street";
            document.Form.Fields["topmostSubform[0].Page1[0].Address[0].f1_8[0]"].Value = "Washington, DC  22222";
            document.Form.Fields["topmostSubform[0].Page1[0].f1_9[0]"].Value = "Any Requester";
            document.Form.Fields["topmostSubform[0].Page1[0].f1_10[0]"].Value = "17288825617";



            string ssn = "265748".Replace("-", "").Replace(" ", "");
            string ein = "521234567".Trim().Replace("-", "").Replace(" ", "");
            if (ssn.Length >= 9)
            {
                document.Form.Fields["topmostSubform[0].Page1[0].SSN[0].f1_11[0]"].Value = ssn.Substring(0, 1) + ssn.Substring(1, 1) + ssn.Substring(2, 1);
                document.Form.Fields["topmostSubform[0].Page1[0].SSN[0].f1_12[0]"].Value = ssn.Substring(3, 1) + ssn.Substring(4, 1);
                document.Form.Fields["topmostSubform[0].Page1[0].SSN[0].f1_13[0]"].Value = ssn.Substring(5, 1) + ssn.Substring(6, 1) + ssn.Substring(7, 1) + ssn.Substring(8, 1);
            }
            else if (ein.Length >= 9)
            {
                document.Form.Fields["topmostSubform[0].Page1[0].EmployerID[0].f1_14[0]"].Value = ein.Substring(0, 1) + ein.Substring(1, 1);
                document.Form.Fields["topmostSubform[0].Page1[0].EmployerID[0].f1_15[0]"].Value = ein.Substring(2, 1) + ein.Substring(3, 1) + ein.Substring(4, 1) + ein.Substring(5, 1) + ein.Substring(6, 1) + ein.Substring(7, 1) + ein.Substring(8, 1);
            }
            // Uncomment to make form read only
            //document.Form.IsReadOnly = true;

            // Outputs the W-9 to the current web page
            document.Draw(outputPdfPath);

        }
    }
}
